### PR TITLE
[SDL-0236] Update mismatch in TireStatus structure 

### LINKED ISSUE #1089: [SDL 0236] Update mismatch in TireStatus structure
Proposal: [Update mismatch in TireStatus structure](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0236-TireStatus-Mismatch.md)

> This proposal is about correcting a mismatch between HMI_API and MOBILE_API for the `TireStatus` structure.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/743

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revision: clarify that this update will be a breaking change.

> The proposal .md file was updated to reflect these revisions on 6/6/19.

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Fixes  #1089 

This PR is **[ready]** for review.

### Risk
This PR makes **[major]** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Core Tests
TBD

Core version / branch / commit hash / module tested against: https://github.com/LuxoftSDL/sdl_core/tree/feature/sdl_0236_tireStatus_mismatch
HMI name / version / branch / commit hash / module tested against: develop

### Summary
Applied [SDL-0236] Update mismatch in TireStatus structure changes

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
Copy of https://github.com/smartdevicelink/sdl_java_suite/pull/1577

**Comment by YuriiLokhmatovGitHub** [comment]
@LitvinenkoIra, could you please approve the version change commit (When a first-time contributor submits a pull request to a public repository, a maintainer with write access must approve any workflow runs.)

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/GetVehicleDataResponseTests.java
Comment (atiwari9) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/GetVehicleDataResponseTests.java:148]: Please remove the blank newline as this file does not have any other changes.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java
Comment (atiwari9) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java:245]: So getters like these now could possibly return null objects right? Is there a place where this can be notified to the developer to be cautious?

Comment (YuriiLokhmatovGitHub) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java]: Added the @Nullable modifier.

Comment (YuriiLokhmatovGitHub) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java]: Added text comment: 'Optional since SmartDeviceLink 8.0.0'

**Comment by YuriiLokhmatovGitHub** [comment]
Hello @atiwari9. I have updated the PR, could you please review the changes? Thank you.

**Review by atiwari9 — APPROVED** [review]

**Comment by LitvinenkoIra** [comment]
@theresalech This PR is ready for Livio review. Thank you!

**Review by JulianKast — CHANGES_REQUESTED** [review]
Hello, To avoid a breaking change, we need to do a workaround for the case of null and issue a warning until we do a major version change. I have requested changes for all of the SingleTireStatus getters already, in addition to that, could you modify the getter for pressureTellTale to look like:

```
    public WarningLightStatus getPressureTelltale() {
        WarningLightStatus warningLightStatus = (WarningLightStatus) getObject(WarningLightStatus.class, KEY_PRESSURE_TELLTALE);
        if (warningLightStatus == null) {
            WarningLightStatus newWarningLightStatus = WarningLightStatus.NOT_USED;
            setValue(KEY_PRESSURE_TELLTALE, newWarningLightStatus);
            warningLightStatus = newWarningLightStatus;
            DebugTool.logWarning(TAG, "TireStatus.pressureTelltale was null and will be set to .notUsed. In the future, this will change to be nullable.");
        }
        return warningLightStatus;
    }
```

also, could you remove the @NonNull from the setter for PressureTelltale and add a TAG to this class for the warnings

```
    private static final String TAG = "TireStatus";
```

**Comment by YaroslavLutsenko** [comment]
Hello @JulianKast ! I've processed these changes, please take a look at this (9d2353d)

**Review by JulianKast — APPROVED** [review]

**Pull request merged**
